### PR TITLE
feat: implement EthGetBlockReceiptsLimited RPC method

### LINKED ISSUE #5017: `Filecoin.EthGetBlockReceiptsLimited`

### PR DESCRIPTION
**Pull request by virajbhartiya**
## Summary of changes

Changes introduced in this pull request:

implemented EthGetBlockReceiptsLimited RPC method
-

## Reference issue to close (if applicable)
Closes #5017

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation. All new code adheres to the team's [documentation standards](https://github.com/ChainSafe/forest/wiki/Documentation-practices),
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/tool/subcommands/api_cmd.rs [hunk 1]
```diff
@@ -1441,7 +1441,7 @@ fn eth_tests_with_tipset<DB: Blockstore>(store: &Arc<DB>, shared_tipset: &Tipset
         ),
         RpcTest::identity(EthGetBlockReceipts::request((block_hash.clone(),)).unwrap()),
         RpcTest::identity(
-            EthGetBlockTransactionCountByHash::request((block_hash.clone(),)).unwrap(),
+            EthGetBlockReceiptsLimited::request((block_hash.clone(), EthUint64(800))).unwrap(),
```
Comment (elmattic) [inline src/tool/subcommands/api_cmd.rs:1444]: Do not remove existing tests.

**Review by elmattic — CHANGES_REQUESTED** [review]

File: src/rpc/methods/eth.rs [hunk 2]
```diff
@@ -1330,6 +1330,59 @@ impl RpcMethod<1> for EthGetBlockReceipts {
...
+            )?;
+
+            let tx_receipt = new_eth_tx_receipt(&ctx, &tx, &message_lookup).await?;
+            receipts.push(tx_receipt);
+        }
+        Ok(receipts)
+    }
+}
+
+
```
Comment (elmattic) [inline src/rpc/methods/eth.rs:1384]: remove newlines: Format on save?

File: src/tool/subcommands/api_cmd.rs (hunk 1 above)
Comment (virajbhartiya) [reply src/tool/subcommands/api_cmd.rs:1447]: Oh sorry, I was cloning the line instead it got removed

File: src/rpc/methods/eth.rs (hunk 2 above)
Comment (virajbhartiya) [reply src/rpc/methods/eth.rs]: Fixed it

**Comment by elmattic** [comment]
Also don't duplicate all the method body: `EthGetBlockReceipts` and `EthGetBlockReceiptsLimited` should share common code.

Check out how this is done for other `XXXLimited` methods.

**Review by elmattic — APPROVED** [review]

File: src/rpc/methods/eth.rs [hunk 3]
```diff
@@ -1281,6 +1281,51 @@ impl RpcMethod<2> for EthGetBlockByNumber {
     }
 }
 
+async fn get_block_receipts<DB: Blockstore + Send + Sync + 'static>(
+    ctx: &Ctx<DB>,
+    block_hash: EthHash,
+    limit: Option<u64>,
```
Comment (hanabi1224) [inline src/rpc/methods/eth.rs:1287]:
```suggestion
    limit: Option<usize>,
```

File: src/rpc/methods/eth.rs [hunk 4]
```diff
@@ -1281,6 +1281,51 @@ impl RpcMethod<2> for EthGetBlockByNumber {
...
+) -> Result<Vec<EthTxReceipt>, ServerError> {
+    let ts = get_tipset_from_hash(ctx.chain_store(), &block_hash)?;
+    let ts_ref = Arc::new(ts);
+    let ts_key = ts_ref.key();
+    let (state_root, msgs_and_receipts) = execute_tipset(ctx, &ts_ref).await?;
+
+    let msgs_and_receipts = if let Some(limit) = limit {
+        msgs_and_receipts.into_iter().take(limit as usize).collect()
```
Comment (hanabi1224) [inline src/rpc/methods/eth.rs:1295]:
```suggestion
        msgs_and_receipts.into_iter().take(limit).collect()
```

File: src/rpc/methods/eth.rs [hunk 5]
```diff
@@ -1295,38 +1340,25 @@ impl RpcMethod<1> for EthGetBlockReceipts {
...
+    const API_PATHS: ApiPaths = ApiPaths::V1;
+    const PERMISSION: Permission = Permission::Read;
+    type Params = (EthHash, EthUint64);
+    type Ok = Vec<EthTxReceipt>;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (block_hash, EthUint64(limit)): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        get_block_receipts(&ctx, block_hash, Some(limit)).await
```
Comment (hanabi1224) [inline src/rpc/methods/eth.rs:1361]:
```suggestion
        get_block_receipts(&ctx, block_hash, Some(limit as usize)).await
```

**Comment by elmattic** [comment]
Also, remember to include the issue number being closed after Closes in the Reference issue to close (if applicable) section.

**Comment by virajbhartiya** [comment]
Updated the description

**Review by hanabi1224 — APPROVED** [review]

**Pull request merged**